### ISSUE TITLE
[prometheus][remote_write] Failing to parse some histogram fields

### ISSUE THREAD

**Issue by tetianakravchenko** [elastic/integrations#7893]
Some documents are dropped due to:

```
"prometheus\":{\"apiserver_flowcontrol_priority_level_request_utilization\":{\"histogram\":{\"counts\":[5000945144],\"values\":[0.25]}},\"labels\":{\"instance\":\"10.128.0.10:443\",\"job\":\"kubernetes-apiservers\",\"phase\":\"waiting\",\"priority_level\":\"node-high\"}},\"service\":{\"type\":\"prometheus\"}}, Private:interface {}(nil), TimeSeries:true}, Flags:0x0, Cache:publisher.EventCache{m:mapstr.M(nil)}} (status=400): {\"type\":\"document_parsing_exception\",
    
    \"reason\":\"[1:2472] failed to parse field [prometheus.apiserver_flowcontrol_priority_level_request_utilization.histogram] of type [histogram]\",\"caused_by\":{\"type\":\"illegal_argument_exception\",
    
    "reason\":\"[1:2482] Numeric value (5000945144) out of range of int (-2147483648 - 2147483647)\\n at 
```

```
"reason":"[1:3039] failed to parse field [prometheus.go_gc_pauses_seconds_total.histogram] of type [histogram]","caused_by":{"type":"document_parsing_exception","reason":"[1:3039] error parsing field [prometheus.go_gc_pauses_seconds_total.histogram], [values] values must be in increasing order, got [-4.9E-324] but previous value was [0.0]"}}, dropping event!
```

This could be related to the fact that the datastream was actually dropped first to empty the index

**Comment by tetianakravchenko** [comment]
The second error (`[values] values must be in increasing order, got [-4.9E-324] but previous value was [0.0]"`) is related to this issue - https://github.com/elastic/beats/issues/36317, and is going to be fixed soon.

**Comment by pjbertels** [comment]
'kubernetes-apiservers' and job_name: 'kubernetes-cadvisor' are the two scraping target that generate the histograms in my setup.

**Comment by tetianakravchenko** [comment]
I was able to reproduce the issue on my setup as well for multiple `apiserver_flowcontrol_*` histograms, it is actually just 3 metrics: `apiserver_flowcontrol_priority_level_request_utilization`,
`apiserver_flowcontrol_demand_seats`,
`apiserver_flowcontrol_read_vs_write_current_requests`

After some time, I see the histogram metric - `prometheus.apiserver_flowcontrol_priority_level_request_utilization.histogram`, but it is empty - `{"values":[],"counts":[]}`, not sure if it is a correct value:
<img width="630" alt="Screenshot 2023-09-22 at 14 24 02" src="https://github.com/elastic/integrations/assets/28299531/960e1544-fc84-4b52-9655-bacbe0747fb6">

**Comment by tetianakravchenko** [comment]
opened elasticsearch issue - https://github.com/elastic/elasticsearch/issues/99820

**Comment by tetianakravchenko** [comment]
first error - `Numeric value (5000945144) out of range of int (-2147483648 - 2147483647)` should be fixed in https://github.com/elastic/elasticsearch/issues/99820

second error - `[values] values must be in increasing order, got [-4.9E-324] but previous value was [0.0]"` should be closed in https://github.com/elastic/beats/pull/36647

both PRs were merged and will be available in 8.11.0
